libs/community/langchain_community/document_loaders/sqlalchemy.py

```diff
@@ -0,0 +1,144 @@
+import functools
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import sqlalchemy as sa
+
+from langchain_community.docstore.document import Document
+from langchain_community.document_loaders.base import BaseLoader
+
+
+class SQLAlchemyLoader(BaseLoader):
+    """
+    Load documents by querying database tables supported by SQLAlchemy.
+    Each document represents one row of the result.
+    """
+
+    def __init__(
+        self,
+        query: Union[str, sa.Select],
+        url: str,
+        page_content: Optional[Callable] = None,
+        metadata: Optional[Callable] = None,
```

Comment:
The name seems to suggest that we're passing the page_content and metadata themselves.
Can we use a name that shows that it's a function ? For instance `page_content_mapper` and `metadata_mapper`. Or something better if you have an idea.